src/aiperf/workers/worker.py

```diff
@@ -194,8 +211,18 @@ async def _execute_single_credit_internal(
 
         turn_list = []
         for turn_index in range(len(conversation.turns)):
-            self.task_stats.total += 1
+            # Apply turn delay BEFORE sending the turn (simulating user thinking time)
+            # Skip delay for the first turn
             turn = conversation.turns[turn_index]
+            if turn_index > 0 and turn.delay is not None and turn.delay > 0:
+                delay_seconds = turn.delay / 1000.0  # Convert milliseconds to seconds
```

Comment:
Use `MILLIS_PER_SECOND` from `aiperf.common.constants` for self documenting code